process/sync/shardblock.go

```diff
@@ -245,7 +320,50 @@ func (boot *ShardBootstrap) removeBlockBody(
 	return nil
 }
 
-func (boot *ShardBootstrap) getNonceWithLastNotarized(nonce uint64) (uint64, map[uint32]uint64, map[uint32]uint64) {
+// find highest nonce metachain which notarized a shardheader - as finality is given by metachain
+func (boot *ShardBootstrap) getHighestHdrForOwnShardFromMetachain(
+	startOwnNonce uint64,
+	lastMetaHdrNonce uint64,
+) (uint64, map[uint32]uint64, map[uint32]uint64) {
+
+	metaHdrNonce := lastMetaHdrNonce
+
+	for {
+		if metaHdrNonce < 1 {
```

Comment:
maybe replace magic number with a constant

Reply:
not really a magic number, but changed.